packages/hawtio/src/plugins/camel/CamelPreferences.tsx

```diff
@@ -0,0 +1,111 @@
+import { CardBody, Checkbox, Form, FormGroup, FormSection, TextInput } from '@patternfly/react-core'
+import React, { useState } from 'react'
+import { camelPreferencesService, ICamelPreferences } from './camel-preferences-service'
+
+export const CamelPreferences: React.FunctionComponent = () => (
+  <CardBody>
+    <Form isHorizontal>
+      <CamelPreferencesForm />
+    </Form>
+  </CardBody>
+)
+
+const CamelPreferencesForm: React.FunctionComponent = () => {
+  const [camelPreferences, setCamelPreferences] = useState(camelPreferencesService.loadCamelPreferences())
+
+  const updatePreferences = (value: boolean | number, key: keyof ICamelPreferences): void => {
+    const updatedPreferences = { ...camelPreferences, ...{ [key]: value } }
```

Comment:
Preferring this method to camelPreferences[key] = value to ensure there is a new pointer for every update, so useState updates properly